app/src/main/cpp/skyline/kernel/memory.cpp

```diff
@@ -7,13 +7,165 @@
 #include "types/KProcess.h"
 
 namespace skyline::kernel {
-    MemoryManager::MemoryManager(const DeviceState &state) : state(state) {}
+    MemoryManager::MemoryManager(const DeviceState &state) noexcept : state(state), setHeapSize(), chunks() {}
 
-    MemoryManager::~MemoryManager() {
+    MemoryManager::~MemoryManager() noexcept {
         if (base.valid() && !base.empty())
             munmap(reinterpret_cast<void *>(base.data()), base.size());
     }
 
+    std::map<u8 *,ChunkDescriptor>::iterator MemoryManager::upper_bound(u8 *address) {
+        std::map<u8 *,ChunkDescriptor>::iterator result{chunks.begin()};
```

Comment:
uh, why can you not use regular upper bound here. Doing it this way seems to negate all the benefits of using an rbtree no?

Reply:
Uh I'd try and look into why that's the case then, cause using if you're using a linear search for a tree you may as well use a vector (it's fine if you use that too)